File: Core/Domain/Ingredient.cs
using System;

namespace Codell.Pies.Core.Domain
{
    public class Ingredient
    {
        public Ingredient(Guid id, string description, int percent, string color)
        {
            Id = id;
            Percent = percent;
            Description = description;
            Color = color;
        }

        public Guid Id { get; set; }

        public string Description { get; set; }

        public string Color { get; set; }

        public int Percent { get; set; }
    }
}

File: Web/Scripts-Testing/Creators/ingredient.js
var cr8 = cr8 || {};
cr8.Ingredient = function (id) {
    var self = this;
    self.id = !id ? 'x' : id;
    self.percent = 20;
    self.description = self.id + ' description';
    self.color = "blue";
    self.matches = function (other) {
        if (!other) return false;
        return self.id == other.id &&
               (self.percent == other.percent || self.percent == other.percent())  &&
               (self.description == other.description || self.description == other.description()) &&
               (self.color == other.color || self.color == other.color());
    };
    return self;
};

cr8.IngredientsUpdated = function () {
    var self = this;
    self.ingredients = [new cr8.Ingredient()];
    self.filler = new cr8.Ingredient('filler');
    self.withoutFiller = function () {
        self.filler.percent = 0;
        return self;
    };
    return self;
};
